atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/KvTableMappingService.java

```diff
@@ -134,22 +134,20 @@ public TableReference addTable(TableReference tableRef) {
             return tableRef;
         }
 
-        TableReference cachedShortName = tableMap.get().get(tableRef);
-        if (cachedShortName != null) {
-            logDebugOrTrace(
-                    "table mapping already exists",
-                    LoggingArgs.tableRef("longTableRef", tableRef),
-                    SafeArg.of("shortTableRef", cachedShortName));
-            return cachedShortName;
-        }
-
+        // In the past, a lookup in the table cache (tableMap) would be performed when adding a new table.
```

Comment:
I'm not sure this is going to work. Because table mappings aren't deterministic (We just use the long supplier to generate the short name), a previously mapped long to short name won't work correctly. In practice the long supplier is a timestamp service which'll punch new timestamp values...

For example:

- Node 1 creates a table, stores to the namespace table, assigns key: long table ref, value: timestamp 123
- Node 2 creates a table, stores to the namespace table, but this time assigns key long table ref, value: timestamp 124
- Now node 1 and node 2 disagree about which table the long table reference points to

The question is why doesn't this fail during TableRemappingKeyValueServiceTest? I will try and write a test to exercise this path and see what happens....

Reply:
Ah, no, the changing bit (The timestamp) is in the value - the key remains the same, and a KeyAlreadyExistsException will be thrown if putUnlessExists() is called with the same key (The long table name) and value (The short name). So I think we're OK here.